src/plugins/geomap/geomap.js

```diff
@@ -22,18 +22,30 @@ var componentName = "wb-geomap",
 		// returns DOM object = proceed with init
 		// returns undefined = do not proceed with init (e.g., already initialized)
 		var elm = wb.init( event, componentName, selector ),
-			$elm, modeJS;
+			$elm, $map, $leg, modeJS;
 
 		if ( elm ) {
 			$elm = $( elm );
+			$map = $elm.find( ".wb-geomap-map" );
+			$leg = $elm.find( ".wb-geomap-legend" );
+
+			// Initialize the map height for the loader
+			$map.height( $map.width() * 0.8 );
+			$map.append( "<div class='geomap-progress'><span class='wb-inv'>" + wb.i18n( "load" ) + "</span></div>" );
+			$leg.append( "<div style='background-color:#f5f5f5;width:100%;height:25px;margin-top:10px;'><span style='display:block;background-color:#fff;margin-left:25px;width:15px;height:25px;'></span></div>" +
+					"<div style='background-color:#f5f5f5;width:100%;height:25px;margin:20px 0 20px 0'><span style='display:block;background-color:#fff;margin-left:25px;width:15px;height:25px;'></span></div>" +
```

Comment:
Purely for UX. They are interface skeletons. Without it we would probably want a spinner but since the map element already as one it would be confusing to have two spinners. So we have a skeleton in the legend element while the map is loading.